src/server/response.cpp

```diff
@@ -275,14 +274,12 @@ MHD_Result Response::send(const RequestContext& request, MHD_Connection* connect
 {
   MHD_Response* response = create_mhd_response(request);
 
-  if ( m_mode != ResponseMode::ERROR_RESPONSE ) {
-    MHD_add_response_header(response, "Access-Control-Allow-Origin", "*");
-    MHD_add_response_header(response, MHD_HTTP_HEADER_CACHE_CONTROL,
-      m_etag.get_option(ETag::CACHEABLE_ENTITY) ? "max-age=2723040, public" : "no-cache, no-store, must-revalidate");
-    const std::string etag = m_etag.get_etag();
-    if ( ! etag.empty() )
-        MHD_add_response_header(response, MHD_HTTP_HEADER_ETAG, etag.c_str());
-  }
+  MHD_add_response_header(response, "Access-Control-Allow-Origin", "*");
+  MHD_add_response_header(response, MHD_HTTP_HEADER_CACHE_CONTROL,
+    m_etag.get_option(ETag::CACHEABLE_ENTITY) ? "max-age=2723040, public" : "no-cache, no-store, must-revalidate");
+  const std::string etag = m_etag.get_etag();
+  if ( ! etag.empty() )
+    MHD_add_response_header(response, MHD_HTTP_HEADER_ETAG, etag.c_str());
```

Comment:
Now I see what you meant in your question about excluded headers. In the morning I had looked in the wrong place (`Response::create_error_response()` and its call site). Should have read your question twice. I definitely made this change intentionally, but I am not sure if it is really required or not. At least my unit test for 416 responses doesn't validate the absence of those headers.